### ISSUE TITLE
Improve our documentation for contributors

### ISSUE THREAD

**Issue by smalls** [GoogleWebComponents/model-viewer#279]
We have some missing documentation that we worry may make it harder for contributors than it needs to be. In order to be a transparent project, let's clean this up and give potential contributors the documentation that they need to be successful.

Particular problems areas that we worry about:

- [ ] More architecture and design documentation. In particular there are some places in the code where the eventual architecture may not yet be apparent, so can we make that clear in documentation?
- [ ] More emphasis around our testing strategy and where we leverage unit tests (and how to run them)
- [ ] Any clarity we can add in CONTRIBUTING.md and the various templates - including taking a pass through to make sure they're coherent.

**Comment by elalish** [comment]
@cdata I think we said this is part of 1.0, right?

**Comment by cdata** [comment]
Yeah, I would say all three bullets are not satisfactorily addressed.
